front-packages/akeneo-design-system/src/components/Tiles/Tiles.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {Ref, ReactNode, isValidElement} from 'react';
+import styled, {css} from 'styled-components';
+import {IconProps} from '../../icons';
+import {AkeneoThemedProps, getColor} from '../../theme';
+
+type Size = 'small' | 'big';
+
+const TilesContainer = styled.div<{size: Size} & AkeneoThemedProps>`
+  display: grid;
+  ${({size}) =>
+    size === 'small'
+      ? css`
+          gap: 20px;
+          grid-template-columns: repeat(auto-fill, minmax(130px, 1fr));
+        `
+      : css`
+          gap: 30px;
+          grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
+        `}
+`;
+
+const TileContainer = styled.div<{selected: boolean; size: Size; onClick?: () => void} & AkeneoThemedProps>`
+  ${({size}) =>
+    size === 'small'
+      ? css`
+          height: 130px;
+        `
+      : css`
+          height: 200px;
+        `}
+  transition: border-color 0.2s, color 0.2s, background 0.2s;
+  ${({onClick}) =>
+    onClick !== undefined &&
+    css`
+      cursor: pointer;
+    `}
+  text-align: center;
+  ${({selected}) =>
+    selected
+      ? css`
+          border: 2px solid ${getColor('blue', 100)};
+          color: ${getColor('blue', 100)};
+          margin: -1px;
+          background: ${getColor('blue', 10)};
+        `
+      : css`
+          border: 1px solid ${getColor('grey', 80)};
+        `}
+  &:hover {
+    border: 2px solid ${getColor('blue', 100)};
+    color: ${getColor('blue', 100)};
+    margin: -1px;
+    background: ${getColor('blue', 10)};
+  }
+`;
+
+const IconContainer = styled.div<{size: Size} & AkeneoThemedProps>`
+  ${({size}) =>
+    size === 'small'
+      ? css`
+          padding: 25px 0 0 0;
+          height: 54px;
+        `
+      : css`
+          padding: 40px 0 0 0;
+          height: 100px;
+        `}
+`;
+const LabelContainer = styled.div`
+  margin: 10px;
+  display: -webkit-box;
+  -webkit-line-clamp: 2;
+  -webkit-box-orient: vertical;
+  overflow: hidden;
+  line-height: 1.3;
+`;
+
+type TilesProps = {
+  /**
+   * Children are Tile components only
+   */
+  children?: ReactNode;
+
+  /**
+   * The size can be 'small' (by default) or 'big'
+   */
+  size?: Size;
+};
+
+type TileProps = {
+  label: string;
+  icon: React.ReactElement<IconProps>;
+  size?: Size;
+  selected?: boolean;
+  onClick?: () => void;
+};
+
+const Tile: React.FC<TileProps> = ({label, icon, selected = false, size, onClick, ...rest}) => {
```

Comment:
```suggestion
const Tile: FC<TileProps> = ({label, icon, selected = false, size, onClick, ...rest}) => {
```

We tend to avoid referencing `React.`